lib/rucio/core/rule.py

```diff
@@ -1910,9 +2203,32 @@ def update_rules_for_lost_replica(scope, name, rse_id, nowait=False, *, session:
     :param logger:         Optional decorated logger that can be passed from the calling daemons or servers.
     """
 
-    locks = session.query(models.ReplicaLock).filter(models.ReplicaLock.scope == scope, models.ReplicaLock.name == name, models.ReplicaLock.rse_id == rse_id).with_for_update(nowait=nowait).all()
-    replica = session.query(models.RSEFileAssociation).filter(models.RSEFileAssociation.scope == scope, models.RSEFileAssociation.name == name, models.RSEFileAssociation.rse_id == rse_id).with_for_update(nowait=nowait).one()
-    requests = session.query(models.Request).filter(models.Request.scope == scope, models.Request.name == name, models.Request.dest_rse_id == rse_id).with_for_update(nowait=nowait).all()
+    stmt = select(
+        models.ReplicaLock
+    ).where(
+        and_(models.ReplicaLock.scope == scope,
+             models.ReplicaLock.name == name,
+             models.ReplicaLock.rse_id == rse_id)
+    ).with_for_update(nowait=nowait)
+    locks = session.execute(stmt).all()
```

Comment:
You’ve forgotten a `scalars()` here. Same for the third query.

Reply:
Fixed, now:

`locks = session.execute(stmt).scalars().all()`
`requests = session.execute(stmt).scalars().all()`